internal/testrunner/runners/system/servicedeployer/terraform.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package servicedeployer
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/files"
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/stack"
+)
+
+// TerraformServiceDeployer is responsible for deploying infrastructure described with Terraform definitions.
+type TerraformServiceDeployer struct {
+	definitionsDir string
+}
+
+// NewTerraformServiceDeployer creates an instance of TerraformServiceDeployer.
+func NewTerraformServiceDeployer(definitionsDir string) (*TerraformServiceDeployer, error) {
+	return &TerraformServiceDeployer{
+		definitionsDir: definitionsDir,
+	}, nil
+}
+
+// SetUp method boots up the Docker Compose with Terraform executor and mounted .tf definitions.
+func (tsd TerraformServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedService, error) {
+	logger.Debug("setting up service using Terraform service deployer")
+	terraformDeployerYml, err := install.ServiceDeployerComposeFile("terraform")
+	if err != nil {
+		return nil, errors.Wrap(err, "can't locate docker compose file for service deployer")
+	}
+
+	service := dockerComposeDeployedService{
+		ymlPath: terraformDeployerYml,
+		project: "elastic-package-service",
+	}
+	outCtxt := inCtxt
+
+	p, err := compose.NewProject(service.project, service.ymlPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create docker compose project for service")
+	}
+
+	// Clean service logs
+	err = files.RemoveContent(outCtxt.Logs.Folder.Local)
+	if err != nil {
+		return nil, errors.Wrap(err, "removing service logs failed")
+	}
+
+	// Boot up service
+	tfEnvironment := tsd.buildTerraformExecutorEnvironment(inCtxt)
+	opts := compose.CommandOptions{
+		Env:       tfEnvironment,
+		ExtraArgs: []string{"--build", "-d"},
+	}
+	if err := p.Up(opts); err != nil {
+		return nil, errors.Wrap(err, "could not boot up service using docker compose")
+	}
+
+	// Build service container name
+	serviceContainer := fmt.Sprintf("%s_terraform_1", service.project)
+	outCtxt.Hostname = serviceContainer
+
+	// Connect service network with stack network (for the purpose of metrics collection)
```

Comment:
Apologies if you explained this to me already a couple days ago, but what is the reason for connecting the terraform executor to the stack network? Will any of the stack containers (e.g. agent) need to talk to the terraform executor container?

Reply:
It was for next use cases, where we may want to proxy traffic from an external endpoint, but as long as I removed ports I think we can remove it too.

I think we'll have to check the progress of research on running the Elastic Agent in the k8s cluster and see how we can combine it with elastic-package.